src/AutoRest.CSharp/Mgmt/AutoRest/MgmtTarget.cs

```diff
@@ -40,31 +41,31 @@ public static void Execute(GeneratedCodeWorkspace project, CodeModel codeModel,
                 project.AddGeneratedFile($"Models/{name}.Serialization.cs", serializerCodeWriter.ToString());
             }
 
-            foreach (var client in context.Library.RestClients)
+            foreach (var client in context.Library.RestClients.Where(client => !context.Library.FindOperationGroup(client).ShouldSkipGeneration(configuration.MgmtConfiguration)))
```

Comment:
Instead of filtering during the writing, we should avoid building these models altogether.